airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -568,14 +588,19 @@ def generate_final_model(self, from_table: str, column_names: Dict[str, Tuple[st
     def list_fields(self, column_names: Dict[str, Tuple[str, str]]) -> List[str]:
         return [column_names[field][0] for field in column_names]
 
-    def add_to_outputs(self, sql: str, is_intermediate: bool, suffix: str = "") -> str:
+    def add_to_outputs(self, sql: str, is_intermediate: bool, column_count: int = 0, suffix: str = "") -> str:
         schema = self.get_schema(is_intermediate)
         table_name = self.generate_new_table_name(is_intermediate, suffix)
         file_name = self.get_file_name(self.stream_name, schema, table_name)
         self.add_table_to_local_registry(file_name, table_name, is_intermediate)
         file = f"{file_name}.sql"
         if is_intermediate:
-            output = os.path.join("airbyte_ctes", self.schema, file)
+            if column_count <= MAXIMUM_COLUMNS_TO_USE_EPHEMERAL:
+                output = os.path.join("airbyte_ctes", self.schema, file)
+            else:
+                # dbt throws "maximum recursion depth exceeded" exception at runtime
```

Comment:
for my understanding: in this case we'll end up using a mix of ephemeral and views to create the final table?

Reply:
yes